Tools/debug/CollectSystemInfo.ps1

```diff
@@ -0,0 +1,214 @@
+        Write-Warning "Failed to collect Windows Update logs: $_"
+    }
+}
+
+function Get-WindowsUptime {
+    try {
+        $uptime = (Get-Date) - (gcim Win32_OperatingSystem).LastBootUpTime
+        $uptime.ToString() | Out-File -FilePath 'WindowsUptime.txt'
+        Write-Host 'Windows uptime collection completed.'
+    } catch {
+        Write-Warning "Failed to collect Windows uptime: $_"
+    }
+}
+
+function Get-ServicesList {
+    try {
+        Get-Service | Select-Object -Property Name, DisplayName, Status, StartType | Export-Csv -Path 'Services.csv' -NoTypeInformation
+    } catch {
+        Write-Warning "Failed to collect list of services: $_"
+    }
+}
+
+function Get-RunningProcesses {
+    try {
+        Get-Process | Select-Object -Property Id, ProcessName, StartTime | Export-Csv -Path 'RunningProcesses.csv' -NoTypeInformation
+    } catch {
+        Write-Warning "Failed to collect list of running processes: $_"
+    }
+}
+
+function Get-InstalledApplications {
+    try {
+        Get-ItemProperty -Path 'HKLM:\Software\Microsoft\Windows\CurrentVersion\Uninstall\*' |
+        Select-Object -Property DisplayName, DisplayVersion, Publisher, InstallDate |
+        Export-Csv -Path 'InstalledApplications.csv' -NoTypeInformation
+    } catch {
+        Write-Warning "Failed to collect list of installed applications: $_"
+    }
+}
+
+function Get-InstalledKBs {
+    try {
+        Get-HotFix | Select-Object -Property Description, HotFixID, InstalledOn | Export-Csv -Path 'InstalledKBs.csv' -NoTypeInformation
+    } catch {
+        Write-Warning "Failed to collect list of installed KBs: $_"
+    }
+}
+
+function Get-WindowsMemoryDump {
+    $dumpFiles = @()
+    $memoryDumpPaths = @("$env:SystemRoot\MEMORY.DMP", "$env:SystemRoot\Minidump")
+
+    foreach ($path in $memoryDumpPaths) {
+        try {
+            $memoryDumpPath = Get-Item -Path $path -ErrorAction Stop
+            if ($memoryDumpPath) {
+                $dumpFiles += $memoryDumpPath.FullName
+            }
+        } catch {
+            Write-Host "Failed to collect $memoryDumpPath file: $_"
+        }
+    }
+
+    return $dumpFiles
+}
+
+function Remove-TempFiles {
+    param ([string[]]$Path)
+    try {
+        Remove-Item -Path $Path -Recurse -Force
+    } catch {
+        Write-Warning "Failed to delete file: $resolvedFile - $_"
+    }
+}
+
+function Write-InformationToArchive {
+    try {
+        $date = Get-Date -Format 'yyyy-MM-dd_HH-mm-ss'
+        $archiveFile = "SystemInfo_$date.zip"
+        $filesInCurrentDirectory = @('*.txt', '*.csv', '*.evtx', '*.log')
+        $collectedFiles = foreach ($pattern in $filesInCurrentDirectory) {
+            Get-ChildItem -Path $pattern -ErrorAction SilentlyContinue
+        }
+
+        if (Test-Path -Path 'LocaleMetaData') {
+            $collectedFiles += 'LocaleMetaData'
+        }
+
+        if ($IncludeSensitiveData) {
+            $dumpFiles = Get-WindowsMemoryDump
+            $exportedFiles += $dumpFiles
```

Comment:
This naming is the opposite of what I suggested. "Exported" means the data were copied and transformed. "Collected" means just they are grouped together. So it is more appropriate to name "exported" for files we created and "collected" for all archived files.